### PR TITLE
Meta tag diversity

### LINKED ISSUE #61: Meta and OG data should vary based on what page you're on
E.g., on a blog post page, the description should be the blog post teaser and the `og:type` should be `article`.

### PR DESCRIPTION
**Pull request by chigby**
This pull request:

1. Dynamically infers meta/open graph/twitter card tags for blog pages and work pages from their contents.
2. Adds a few meta tags we weren't using before to the base template, namely `og:url` and the site's twitter account.
3. Adds a `meta` block that any page can use to plop more tags into the `<head>` as needed in the future.

Closes #61

### CODE REVIEW

**Comment by melinath** [comment]
I am concerned about this implementation. Right now the `_og_image_url.html` template has to know about every model that might have a teaser image, while `_page_description.html` similarly has to know about every model that might have descriptive metadata.

This should be implemented such that we have a sitewide default, and individual templates can override with their own metadata - i.e. a blog post's metadata should all be in the blog post detail template.

Basically we should have a template block specifically for og / twitter / etc. metadata which is completely overridden (i.e. no `block.super` call) on pages that have more specific information to share.

**Comment by chigby** [comment]
Haha, that's the way I had it but then Harris convinced me to change it to what it is now. I defer to your inevitable consensus @harrislapiroff @melinath

**Comment by harrislapiroff** [comment]
I actually favor having no og image tag in the base template and putting it in each page template same as the content type tags. Overrides should be avoided when possible IMO as should adding too many blocks for specific things. It reduces predictability and increases spaghetti code.

**Comment by harrislapiroff** [comment]
Oh, but we want pages that don't have an image specified to default to the default lw image. Ugh. Okay, I defer to whatever approach.

**Comment by melinath** [comment]
I'm fine with not having a default block in the base template. It sounds like we're both on the same page about what the sub-template talks should look like, though.

**Comment by chigby** [comment]
How's this look? @melinath @harrislapiroff

**Comment by melinath** [comment]
This is definitely WAAAAY clearer to read than the previous version. 💯  Though now that it's split out into separate pages... I would prefer to be able to customize the image and description on a per-page basis. But that's probably a followup issue.

I think I would prefer to have the block called `{% open_graph %}` or `{% metadata %}` or something; I don't think that the open graph information should be lumped in with things like setting the charset & viewport settings. It's fundamentally different. I don't feel... SUPER strongly about this, I guess, but... I kinda do?

**Comment by harrislapiroff** [comment]
I feel like that makes it less predictable and introduces more cases where it's ambiguous what block a line of code belongs in (where do twitter card details go? pinterest rich pin tags? what about just standard description/keyword tags or other metadata like that? It's not immediately clear to me whether those would be in the `meta` block or `open_graph` or `metadata` or why) but I also don't feel super strongly. If you do feel strongly, go for it. ¯\_(ツ)_/¯

**Comment by chigby** [comment]
I kind of agree with Harris here. I don't really have a problem with a giant meta block bucket. All of these tags are already being lumped into the `<meta>` tag anyway. I'm fine with a world where the viewport info, open graph data, charset, apple icon data, stylesheet urls, page title, and so on, are all just considered "metadata" and nothing special beyond that.

**Comment by melinath** [comment]
I want to rename the block even if we don't have a default. What the name should be is an open question. Something flexible enough to make sense for Pinterest, Twitter, and Facebook.

I find this concept ("things used for displaying previews in other sites") to be at least as core to a page as what js files are there, and it's definitely distinct from other concerns. Right now I think it's ambiguous whether the proper spot is meta or extra_head, for example. Having a separate, well-named block for this data would make things clearer.

**Comment by melinath** [comment]
I suppose I'd be fine calling it meta, I just don't want to include ALL the meta tags just because of that name, and I'm concerned that's what would happen.

**Comment by harrislapiroff** [comment]
But the reason other things are separated is because it makes a difference what order they go in...

**Comment by harrislapiroff** [comment]
Not because of conceptual differences

**Comment by melinath** [comment]
Okay, so after hashing this out in slack, here's the conclusion:

1. We're going with grouping all the `<meta>` tags together. Nothing should go in `{% block meta %}` that isn't a `<meta>` tag. (We should probably note that in the template.)
2. The default behavior (i.e. using the site-wide settings for og information) should be split off into an inclusion template for DRY. This is slightly more nested but will ensure that the default implementation remains consistent.

**Comment by chigby** [comment]
@melinath @harrislapiroff Ready for review again.
